packages/validator/src/services/block.ts

```diff
@@ -189,7 +196,12 @@ export class BlockProposingService {
       if (contents === null) {
         (await this.api.beacon.publishBlockV2({signedBlockOrContents: signedBlock, ...opts})).assertOk();
       } else {
-        (await this.api.beacon.publishBlockV2({signedBlockOrContents: {...contents, signedBlock}, ...opts})).assertOk();
+        (
+          await this.api.beacon.publishBlockV2({
+            signedBlockOrContents: {...contents, signedBlock} as SignedBeaconBlockOrContents,
```

Comment:
It's not cast rather type inferring, the TS can't infer the type for `{...contents, signedBlock}` object, because `publishBlockV2` accepts union and for TS is confuse which union item that object belongs to.